### PR TITLE
fix: disabling Touchables if onPress was not set to allow event handling in parent components (#217)

### LINKED ISSUE #217: DrawerItem should have pointerEvents="none" when onPress not set
As in the title, when having DrawerItem embedded within Drawer (react-navigation), I don't want to set `onPress` and let the built-in handlers do their thing. 

Right now, DrawerItem registers Touchable that captures touches and automatically prevents the built-in handler from being called.

Expected behaviour:

I would expect DrawerItem to either not have Touchable component or have pointerEvents none when `onPress` is undefined.

### PR DESCRIPTION
**Pull request by kpsroka**
Fixes #217.

### CODE REVIEW

File: example/src/ButtonExample.js [hunk 1]
```diff
@@ -27,24 +27,33 @@ class ButtonExample extends React.Component<Props, State> {
     };
     const source = require('../assets/chameleon.jpg');
     const { theme: { colors: { background } } } = this.props;
+    const dummyFunction = () => {}; // There's no ripple effect without a valid onPress prop.
```
Comment (ferrannp) [inline example/src/ButtonExample.js]: I would prefer to stick with inline function in all elements like `<Button onPress={(} => {}}>Simple</Button>` if not my brain needs to figure out all the time, wait, what is `dummyFunction` ?

File: src/components/TouchableRipple.js [hunk 2]
```diff
@@ -75,13 +75,16 @@ export default class TouchableRipple extends React.Component<Props, void> {
       ...rest
     } = this.props;
 
+    const disabled = this.props.onPress === undefined;
```
Comment (ferrannp) [inline src/components/TouchableRipple.js]: We need to be able to pass `disabled` as a prop too. This should be: `this.props.disabled || !this.props.onPress`.

File: example/src/ButtonExample.js (hunk 1 above)
Comment (kpsroka) [reply example/src/ButtonExample.js]: Done.

File: src/components/TouchableRipple.js (hunk 2 above)
Comment (kpsroka) [reply src/components/TouchableRipple.js]: Done. I also extracted the disabled prop from the `...rest` spread, so we don't pass it twice.

**Comment by kpsroka** [comment]
PTAL :)

**Pull request merged**